tests/tests/history_api_tests.cpp

```diff
@@ -604,4 +605,194 @@ BOOST_AUTO_TEST_CASE(get_account_history_operations) {
    }
 }
 
+/***
+ * This will create market history. But due to the way 
+ * the server handles it, it will take to long to generate
+ * the data needed. This test should normally be commented
+ * out and used when needed.
+ */
+BOOST_AUTO_TEST_CASE( market_history )
+{
+   class simple_tx
+   {
+      public:
+      simple_tx() {}
+      simple_tx(
+         const graphene::chain::account_object& from, 
+         const graphene::chain::account_object& to, 
+         const graphene::chain::price& amount)
+         : from(from), to(to), amount(amount) {}
+      const graphene::chain::account_object from;
+      const graphene::chain::account_object to;
+      const graphene::chain::price amount;
+   };
+   try {
+      app.enable_plugin("market_history");
+      graphene::app::history_api hist_api( app );
+
+      // create the needed things on the chain
+      ACTORS( (bob) (alice) );
+      transfer( committee_account, alice.id, asset(10000000000) );
+      transfer( committee_account, bob.id  , asset(10000000000) );
+      const graphene::chain::asset_object& usd = create_bitasset( "USD", account_id_type() );
+      // set up feed producers
+      {
+         asset_update_feed_producers_operation op;
+         op.asset_to_update = usd.id;
+         op.issuer = committee_account;
+         op.new_feed_producers = { committee_account, alice.id, bob.id };
+         trx.operations.push_back( op );
+         PUSH_TX( db, trx, ~0 );
+         generate_block();
+         trx.clear();
+         publish_feed( committee_account, asset_id_type(), 2, usd.id, 1, asset_id_type() );
```

Comment:
After generated a block, the `usd` object would become unusable. The same for following code.